server/user.js

```
@@ -624,6 +625,104 @@ exports.deposit = function(req, res, next) {
+     var r =  /^[1-9][0-9]*$/;
+     if (!r.test(amount))
+         return res.render('transfer-request', { user: user, id: uuid.v4(),  warning: 'Not a valid amount' });
+    amount = Math.round(parseFloat(amount) * 100)
+
+    if (!password)
+        return res.render('transfer-request', { user: user,  id: uuid.v4(), warning: 'Must enter a password'});
+
+    database.validateUser(user.username, password, otp, function(err) {
+
+        if (err) {
+            if (err === 'WRONG_PASSWORD')
+                return res.render('transfer-request', { user: user, id: uuid.v4(), warning: 'wrong password, try it again...' });
+            if (err === 'INVALID_OTP')
+                return res.render('transfer-request', { user: user, id: uuid.v4(), warning: 'invalid one-time token' });
+            //Should be an user
+            return next(new Error('Unable to validate user handling transfer: \n' + err));
+        }
+        // Check destination user
+
+        database.getUserFromUsername(destUser,function (err,userData){
+            if(err){
+              if(err="NO_USER"){
+                  return res.render('transfer-request', { user: user, id: uuid.v4(), warning: 'The user you are sending to does not exist' });
+              }
+            }
+            var destUserId=userData.id;
+            var senderUserId=req.user.id;
+            if(senderUserId==destUserId){
+                return res.render('transfer-request', { user: user, id: uuid.v4(), warning: 'You cant send to yourself ...' });
+            }
+
+            database.makeTransfer(uid,senderUserId,destUserId,amount,function(err){
+              if(err){
+                console.log(err);
+                if(err == "NOT_ENOUGH_MONEY"){
+                    return res.render('transfer-request', { user: user, id: uuid.v4(), warning: 'You dont have enough bits to transfer' });
+                }
+                //Should never happen
+                else {
```

Never silently ignore stuff. Log an internal error with the details